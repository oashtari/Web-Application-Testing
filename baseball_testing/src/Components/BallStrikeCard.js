import React, {useState, useEffect} from 'react';

export default function BallStrikeTracker() {
    const [ball, setBall] = useState(0);
    const [strike, setStrike] = useState(0);

    return (
        <div>
        <h2>Balls: {ball}</h2>
        <h2>Strikes: {strike}</h2>
    </div>)
}