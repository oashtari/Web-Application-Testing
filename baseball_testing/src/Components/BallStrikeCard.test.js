import React from 'react';
import { render } from '@testing-library/react';
import BallStrikeCard from './BallStrikeCard';
import renderer from 'react-test-renderer';


test('matches the snapshot', () => {
    const tree = renderer.create(<BallStrikeCard />).toJSON()

    expect(tree).toMatchSnapshot()
});