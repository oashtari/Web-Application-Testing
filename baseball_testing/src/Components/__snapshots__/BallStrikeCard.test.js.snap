// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <h2>
    Balls: 
    0
     
  </h2>
  <h2>
    Strikes: 
    0
     
  </h2>
  <div
    className="all_buttons"
    data-testid="button_container"
  >
    <button
      aria-label={null}
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      Strike
    </button>
    <button
      aria-label={null}
      className="btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      Ball
    </button>
    <button
      aria-label={null}
      className="btn btn-warning"
      onClick={[Function]}
      type="button"
    >
      Foul
    </button>
    <button
      aria-label={null}
      className="btn btn-success"
      onClick={[Function]}
      type="button"
    >
      HIT
    </button>
  </div>
</div>
`;
